src/plugins/intel_cpu/src/transformations/snippets/tpp/pass/brgemm_to_brgemm_tpp.cpp

```diff
@@ -0,0 +1,122 @@
+#include "openvino/pass/pattern/matcher.hpp"
+
+#include <cpu/x64/cpu_isa_traits.hpp>
+
+#include "cpu_shape.h"
+#include "utils/general_utils.h"
+
+
+namespace ov {
+namespace intel_cpu {
+namespace tpp {
+namespace pass {
+
+using namespace snippets::lowered;
+
+namespace {
+template<typename T, typename... Args>
+void set_port_desc(const T& port, Args... params) {
+    PortDescriptorUtils::set_port_descriptor_ptr(port, std::make_shared<PortDescriptor>(params...));
+}
+} // namespace
+
+BrgemmToBrgemmTPP::BrgemmToBrgemmTPP() {
+    MATCHER_SCOPE(BrgemmToBrgemmTPP);
+
+    auto m_brgemm = ov::pass::pattern::wrap_type<snippets::op::Brgemm>();
+
+    auto callback = [=](ov::pass::pattern::Matcher& m) {
+        OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "ov::intel_cpu::pass::BrgemmToBrgemmTPP")
+        const auto node = m.get_match_root();
+        const auto brgemm = ov::as_type_ptr<snippets::op::Brgemm>(node);
+        if (!brgemm || ov::as_type_ptr<tpp::op::BrgemmTPP>(node))
+            OPENVINO_THROW("BrgemmCPU cannot be in body before BrgemmToBrgemmTPP pass");
+
+        if (brgemm->is_dynamic()) {
+            return false;
+        }
+
+        const auto& brgemm_in0_desc = PortDescriptorUtils::get_port_descriptor_ptr(brgemm->input(0));
+        const auto& brgemm_in1_desc = PortDescriptorUtils::get_port_descriptor_ptr(brgemm->input(1));
+        const auto& brgemm_out_desc = PortDescriptorUtils::get_port_descriptor_ptr(brgemm->output(0));
+
+        const auto dimsMatMulIn0 = snippets::utils::get_planar_pshape(brgemm->input(0)).get_shape();
+        const auto dimsMatMulIn1 = snippets::utils::get_planar_pshape(brgemm->input(1)).get_shape();
+
+        const auto M = *++dimsMatMulIn0.rbegin();
+        const auto K = *dimsMatMulIn0.rbegin();
+        const auto N = *dimsMatMulIn1.rbegin();
+
+        const auto element_type_a = brgemm->get_input_element_type(0);
+        const auto element_type_b = brgemm->get_input_element_type(1);
+
+        const auto offset_a = brgemm->get_offset_a();
+        const auto offset_b = brgemm->get_offset_b();
+        const auto offset_c = brgemm->get_offset_c();
+
+        std::shared_ptr<tpp::op::BrgemmTPP> brgemm_tpp = nullptr;
+        if (element_type_a == ov::element::f32) {
+            brgemm_tpp = std::make_shared<tpp::op::BrgemmTPP>(brgemm->input_value(0),
+                                                              brgemm->input_value(1),
+                                                              offset_a, offset_b, offset_c,
+                                                              brgemm_in0_desc->get_layout(),
+                                                              brgemm_in1_desc->get_layout(),
+                                                              brgemm_out_desc->get_layout());
+        }
+        OPENVINO_ASSERT(brgemm_tpp, "DEBUG ASSERT: FAILED TO CREATE BrgemmTPP in the BrgemmToBrgemmTPP pass");
```

Comment:
Fair enough, corrected the message.
Let's leave the assert for now. Since TPP is currently an experimental feature, it is likely that smth went wrong the BrgemmTPP was not created.